### PR TITLE
[test] Add missing types linting for x-grid

### PR DESCRIPTION
**Pull request by oliviertassinari**

### CODE REVIEW

File: packages/grid/x-grid/src/XGrid.test.tsx [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import * as React from 'react';
+// @ts-ignore
```
Comment (dtassone) [inline packages/grid/x-grid/src/XGrid.test.tsx:2]: why do we need this one here?

File: packages/grid/x-grid/package.json [hunk 2]
```diff
@@ -24,6 +24,12 @@
     "datagrid",
     "data-grid"
   ],
+  "scripts": {
+    "precommit": "npm run lint",
+    "lint": "../../../node_modules/.bin/tsc --noEmit && eslint 'src/**/*.{ts,tsx}' --quiet --fix -c ../../../.eslintrc.js && npm run lint:css",
```
Comment (dtassone) [inline packages/grid/x-grid/package.json:29]: Not sure it's still useful the lint script here, if we need to lint from the root folder

File: packages/grid/x-grid/src/XGrid.test.tsx (hunk 1 above)
Comment (oliviertassinari) [reply packages/grid/x-grid/src/XGrid.test.tsx:2]: No idea

File: packages/grid/x-grid/package.json (hunk 2 above)
Comment (oliviertassinari) [reply packages/grid/x-grid/package.json:29]: No strong preferences. We lint eslint from the root but typescript from each package. I personally never run eslint in my local env. I run TypeScript from time to time. Happy to only have eslint at the root.

File: packages/grid/x-grid/src/XGrid.test.tsx (hunk 1 above)
Comment (oliviertassinari) [reply packages/grid/x-grid/src/XGrid.test.tsx:2]: Ok, I have looked at why. The test utils haven't .d.ts type definitions nor written in .ts.

I don't fully understand how this work https://github.com/mui-org/material-ui/blob/980e84b9a2454b3ba72c7562fe0655c6b72f853c/packages/material-ui-lab/src/TabPanel/TabPanel.test.tsx#L3, it seems to be thanks to JSDOC.

**Review by dtassone — APPROVED** [review]

**Pull request merged**